### PR TITLE
[SPARK-14874][SQL][Streaming] Remove the obsolete Batch representation

### PR DESCRIPTION
**Pull request by lw-lin**
## What changes were proposed in this pull request?

The `Batch` class, which had been used to indicate progress in a stream, was abandoned by [[SPARK-13985][SQL] Deterministic batches with ids](https://github.com/apache/spark/commit/caea15214571d9b12dcf1553e5c1cc8b83a8ba5b) and then became useless.

This patch:
- removes the `Batch` class
- ~~does some related renaming~~ (update: this has been reverted)
- fixes some related comments
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by lw-lin** [comment]
@marmbrus @tdas would you mind taking a look? Thanks! :-)

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSink.scala [hunk 1]
```diff
@@ -47,7 +47,7 @@ class FileStreamSink(
   private val fileLog = new FileStreamSinkLog(sqlContext, logPath.toUri.toString)
   private val fs = basePath.getFileSystem(sqlContext.sparkContext.hadoopConfiguration)
 
-  override def addBatch(batchId: Long, data: DataFrame): Unit = {
+  override def addData(batchId: Long, data: DataFrame): Unit = {
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSink.scala]: We don't need to change this name.  Its still a batch of data and one of the parameters is still named `batch`.

**Comment by marmbrus** [comment]
It's fine to remove the class, but lets avoid unneeded renaming.

**Comment by lw-lin** [comment]
Sure, so I'm closing this PR since the removal itself is not worthy for committers to process.
@marmbrus thanks for the review!

**Comment by marmbrus** [comment]
To be clear, if there's a completely unused class, I think it's worth the
time to delete it (dead code is confusing for people trying to learn the
code base).

**Comment by lw-lin** [comment]
@marmbrus thanks for the patient reminder!

Since I've reverted the renaming, and I've checked there's no other completely unused class under `o.a.s.sql.execution.streaming` package, maybe this is ready to go (pending tests). So would mind take another look?

**Comment by lw-lin** [comment]
some build issues unrelated to this PR.

Jenkins retest this please

**Comment by lw-lin** [comment]
Jenkins retest this please

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala [hunk 2]
```diff
@@ -88,7 +88,7 @@ class FileStreamSource(
   }
 
   /**
-   * Returns the next batch of data that is available after `start`, if any is available.
```
Comment (lw-lin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala]: This doc should be updated, following `Source.getBatch()`'s doc change from `Returns the next batch of data that is available after start, if any is available.` to `Returns the data that is between the offsets (start, end].`

**Comment by lw-lin** [comment]
just rebase to master to resolve some conflicts

**Comment by marmbrus** [comment]
Thanks, merging to master.

**Pull request closed without merging**